app/src/org/commcare/android/view/EntityView.java

```diff
@@ -454,25 +454,47 @@ private static void removeSpans(Spannable raw) {
                 hints[i] = Integer.parseInt(mHints[i]);
             }
         }
-
-        // Determine how wide to make columns without a specified width
-        int[] widths = new int[hints.length];
-        int sharedBetween = 0;
-        for(int hint : hints) {
-            if(hint != -1) {
-                fullSize -= hint;
-            } else {
-                sharedBetween++;
+        
+        int claimedSpace = 0;
+        int indeterminateColumns = 0;
+        for (int hint : hints) {
+            if (hint != -1) {
+                claimedSpace += hint;
+            }
+            else {
+                indeterminateColumns++;
             }
         }
-        
-        // Set column widths
-        int defaultWidth = sharedBetween == 0 ? 0 : fullSize / sharedBetween;
-        for(int i = 0; i < hints.length; ++i) {
-            widths[i] = hints[i] == -1 ? defaultWidth : hints[i];
+        if (
+            fullSize < claimedSpace + indeterminateColumns
+            || fullSize > claimedSpace && indeterminateColumns == 0

```

Comment:
Please use parenthesis so that we don't have to think about order of operations for || and &&